control-plane/agents/src/bin/core/controller/policies/rebuild_policies.rs

```diff
@@ -95,13 +93,17 @@ impl Policy for SystemPerf {
     }
 }
 
+fn online_child_count_check(nexus: &Nexus) -> i32 {
+    let mut online_child_cnt = 0;
+    for _ in nexus.children.iter().filter(|c| c.state.online()) {
+        online_child_cnt += 1;
+    }
+    online_child_cnt
+}
```

Comment:
You could consider even adding this to Nexus object itself I guess, or just calling the code directly since it can be a one liner like this:
```suggestion
fn online_child_count_check(nexus: &Nexus) -> usize {
    nexus.children.iter().filter(|c| c.state.online()).count()
}
```

Reply:
Sure, this one liner looks good.